test/NuGet.Services.Staging.Test.EndToEnd/StagingClient/StagingClient.cs

```diff
@@ -164,17 +164,36 @@ public StagingClient(Uri stagingServiceUri, ITestOutputHelper logger)
             return JObject.Parse(responseBody);
         }
 
-        public async Task<JObject> Query(string stageId, string queryString)
+        public async Task<JObject> Query(string stageId, string query, bool includePrerelease=false, int skip=0, int take=20)
         {
             _logger.WriteLine($"StagingClient: Query called for stage {stageId}");
 
             JObject index = await Index(stageId);
 
-            string searchEndpoint = index["resources"].Where(x => x["@type"].ToString() == ServiceTypes.SearchQueryService[0]).ToString();
+            string searchEndpoint = index["resources"].First(x => x["@type"].ToString() == ServiceTypes.SearchQueryService[0])["@id"].ToString();

```

Comment:
the validation of the index structure (and other json structures everywhere in this code) is done in UTs. Since this code is part of the E2E tests, I think it's acceptable for it to fail on Null Ref in case of missing values. The traces are rich enough to help us understand the source of failure. 

---
In reply to: [65845079](https://github.com/NuGet/NuGet.Staging/pull/59#discussion_r65845079) [](ancestors = 65845079)